### PR TITLE
PTK: Check if theres a scheduled action before logging warning.

### PR DESCRIPTION
**Pull request by tjcafferkey**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

The previous implementation would log a warning every time the patterns were empty, which could happen frequently due to the asynchronous nature of pattern fetching. This led to:

1. Unnecessary noise in log files
2. Potential performance impact from excessive logging
3. Confusion for users and developers trying to diagnose issues

By only logging when patterns are empty and no fetch is scheduled, we ensure that warnings are only generated in genuinely problematic situations, such as when the pattern fetching mechanism has failed entirely.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Before checking out this branch:**

1. Delete transient `ptk_patterns`
2. Load your homepage
3. Check that the log is present in your log file

**After checking out this branch:**

1. Delete transient `ptk_patterns`
2. Load your homepage
3. Check that the log is not present in your log file

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Blocks/BlockPatterns.php [hunk 1]
```diff
@@ -141,9 +141,12 @@ public function register_ptk_patterns() {
 
 		$patterns = $this->ptk_patterns_store->get_patterns();
 		if ( empty( $patterns ) ) {
-			wc_get_logger()->warning(
-				__( 'Empty patterns received from the PTK Pattern Store', 'woocommerce' ),
-			);
+			// Check if there's a scheduled 'fetch_patterns' action.
```
Comment (gigitux) [inline plugins/woocommerce/src/Blocks/BlockPatterns.php]: The name of the function is quite self-explanatory. I suggest either adding more context to clarify its purpose or removing the comment altogether.

**Review by gigitux — APPROVED** [review]
Thanks for the quick fix! The PR LGTM! I left a minor comment about the code comment, but feel free to merge as it is if you prefer!

File: plugins/woocommerce/src/Blocks/BlockPatterns.php (hunk 1 above)
Comment (tjcafferkey) [reply plugins/woocommerce/src/Blocks/BlockPatterns.php]: Good point, done in https://github.com/woocommerce/woocommerce/pull/51143/commits/a12af10240f2ff08a66a25f23b486d2efc2dcf29

**Comment by RiaanKnoetze** [comment]
Related: https://wordpress.org/support/topic/ptk-pattern-store/

**Pull request merged**